test/hack/e2e_scripts/install_karpenter.sh

```diff
@@ -9,7 +9,6 @@ if (( "$PRIVATE_CLUSTER" == 'true' )); then
   ADDITIONAL_FLAGS="--set .Values.controller.image.repository=$ACCOUNT_ID.dkr.ecr.$REGION.amazonaws.com/karpenter/snapshot/controller --set .Values.controller.image.digest=\"\""
 fi
 
-# Remove service account annotation when dropping support for 1.23
```

Comment:
We removed this comment, but then didn't remove the service account annotation tagging. Should we also remove the line that removes the annotation?

Reply:
We need the service account annotation for private clusters now. So we will not be removing the service account, and we don't need the comment.